ingestion/post_ingestion_checks.py

```diff
@@ -0,0 +1,262 @@
+            if validate_fqn(fqn):
+                database, table_name = parse_database_and_table_names(
+                    manifest["nodes"][node]
+                )
+
+                dataset_urn = mce_builder.make_dataset_urn_with_platform_instance(
+                    name=f"{database}.{table_name}",
+                    platform=PLATFORM,
+                    platform_instance=INSTANCE,
+                    env=ENV,
+                )
+                database_key = mcp_builder.DatabaseKey(
+                    database=database,
+                    platform=PLATFORM,
+                    instance=INSTANCE,
+                    env=ENV,
+                    backcompat_env_as_instance=True,
+                )
+                database_urn = database_key.as_urn()
+                if "dc_display_in_catalogue" in manifest["nodes"][node]["tags"]:
+                    mappings[dataset_urn] = database_urn
+
+    return mappings
+
+
+def _remove_empty_dicts(d):
+    if not isinstance(d, dict):
+        return d
+    cleaned_dict = {}
+    for k, v in d.items():
+        if isinstance(v, dict):
+            nested = _remove_empty_dicts(v)
+            if nested:  # Only add non-empty dictionaries
+                cleaned_dict[k] = nested
+        elif v:  # Only add non-empty values
+            cleaned_dict[k] = v
+    return cleaned_dict
+
+
+def check_is_part_of_relationships(mappings, graph):
+    """
+    this checks whether datasets from cadet have the ispartof relationship
+    to their container. Without this fmd containers appear empty.
+
+    returns a list of datasets where this relationship is missing
+    """
+
+    missing_is_part_of = []
+
+    for dataset in mappings:
+        relations = list(
+            graph.get_related_entities(
+                dataset,
+                ["IsPartOf"],
+                DataHubGraph.RelationshipDirection.OUTGOING,
+            )
+        )
+        if len(relations) == 0:
+            missing_is_part_of.append(dataset)
+    return missing_is_part_of
+
+
+def create_query_input(platform: str) -> dict:
+    query_input = {
+        "input": {
+            "types": [],
+            "facets": ["_entityType", "tags", "domains", "owners"],
+            "orFilters": [
+                {
+                    "and": [
+                        {
+                            "field": "platform",
+                            "condition": "EQUAL",
+                            "values": [f"urn:li:dataPlatform:{platform}"],
+                        }
+                    ]
+                }
+            ],
+            "query": "",
+            "searchFlags": {"maxAggValues": 2000},
+        }
+    }
+    return query_input
+
+
+def parse_count_by_platform_results(result: dict):
+    parsed_result = {
+        field["field"]: field["aggregations"]
+        for field in result["aggregateAcrossEntities"]["facets"]
+    }
+    return parsed_result
+
+
+def counts_by_platform(env: str, platforms: list):
```

Comment:
I would call this `log_counts_by_platform` and replace the print with a log if it works.

also I'd use a dict comp:

```
query_results = {
    platform: parse_count_by_platform_results(
        graph.execute_graphql(count_by_platform_graph_query, create_query_input(platform)))
    for platform in platforms
}
```

Reply:
believe this should be print rather than log to properly set the output of the step in the gha workflow